projects/systelab-components/src/lib/forms/validators/phone-validator.directive.ts

```diff
@@ -0,0 +1,27 @@
+import {FormControl, NG_VALIDATORS,ValidationErrors, Validator} from '@angular/forms';
+import {Directive} from '@angular/core';
+
+export const phoneValidator = (control: FormControl): ValidationErrors | null => {
+  const regExp = /^(\+{1}\d{2,3}\s?[(]{1}\d{1,3}[)]{1}\s?\d+|\+\d{2,3}\s{1}\d+|\d+){1}[\s|-]?\d+([\s|-]?\d+){1,2}$/;
+
```

Comment:
We have a complex validator because there were some correct phones that were marked as not valid.

 '(([+][(]?[0-9]{1,3}[)]?)|([(]?([0]{2})?[0-9]{1,3}[)]?)|\\s*)\\s*[)]?[-\\s\\.]?[(]?[0-9]{0,4}[)]?([-\\s\\.]?[0-9]{2,3})([-\\s\\.]?[0-9]{2,3})([-\\s\\.]?[0-9]{2,3})'

Reply:
Do you have any examples of correct phones that are marked as invalid or any data set?, in order to use them in the unit test and cover that cases. With this improvement you have been helped other projects to enhance its phone Validation (like the one I'm working). Many thanks for the improvement request!

Reply:
const phoneNumberScenarios = [
	{description: 'Number with 9 digits', text: '939999999', expected: true},
	{description: 'International prefix with 00', text: '0034939999999', expected: true},
	{description: 'International prefix with 00 and spaces', text: '0034 93 999 99 99', expected: true},
	{description: 'International prefix with ()', text: '(34)939999999', expected: true},
	{description: 'International prefix with () and space after )', text: '(34) 939999999', expected: true},
	{description: 'International prefix with () and 00', text: '(0034)939999999', expected: true},
	{description: 'International prefix with () and 00 and separator after )', text: '(0034)-939999999', expected: true},
	{description: 'International prefix with +', text: '+34939999999', expected: true},
	{description: 'Number with - separator', text: '93-9999999', expected: true},
	{description: 'Number with . separator', text: '93.9999999', expected: true},
	{description: 'Number with more than one . separator', text: '93.999.99.99', expected: true},
	{description: 'Number with space separator', text: '93 9999999', expected: true},
	{description: 'Number with more than one space separator', text: '93 999 99 99', expected: true},
	{description: 'Italian short number', text: '0225221', expected: true},
	{description: 'Italian international short number', text: '+39 0225221', expected: true},
	{description: 'Italian mobile number', text: '335 784 5058', expected: true},
	{description: 'Italian international mobile number', text: '+39 335 784 5058', expected: true},
	{description: 'Spanish mobile number with spaces', text: '695 21 81 54', expected: true},
	{description: 'Spanish mobile number groups of three', text: '695 218 154', expected: true},
	{description: 'Spanish mobile number no spaces', text: '695218154', expected: true},
	{description: 'Eight digit number', text: '69521815', expected: true},
	{description: 'Strange characters number', text: '@/()*345678', expected: false},
	{description: 'Separators for each number', text: '93 3 9 3 6 6 2 6', expected: false},
	{description: 'Group with only one number', text: '93 393 66 2 6', expected: false},
	{description: 'International with just one number and no parentesis', text: '1-800-955-9525', expected: true},
	{description: 'International with just one number and +', text: '+1 800 955 9525', expected: true},
	{description: 'International with just one number and 00', text: '001 800 955 9525', expected: true},
	{description: 'International with just one number and separators', text: '+1 858-586-9900', expected: true},
	{description: 'International with three number', text: '(351) 93 393 66 26', expected: true},
	{description: 'International with three number and +', text: '+351 93 393 66 26', expected: true},
	{description: 'International with three number and 00', text: '00351 93 393 66 26', expected: true},
	{description: 'International prefix with () and +', text: '(+34)939999999', expected: false},
];